upsolver-ui/src/components/LogoutButton/LogoutButton.jsx

```diff
@@ -0,0 +1,33 @@
+import React, { useContext } from "react";
+import "./LogoutButton.css";
+import { UserContext } from "../../UserContext.js";
+import { ButtonBase, Typography } from "@mui/material";
+import { Navigate } from "react-router-dom";
+
+export default function LogoutButton() {
+  const { user, updateUser } = useContext(UserContext);
+
+  const handleLogout = async () => {
+    try {
+      const response = await fetch("http://localhost:3001/users/logout", {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+          "x-access-token": user?.token,
+        },
+      });
+      if (response.status === 200) {
+        updateUser(null);
+        <Navigate to="/login" replace={true} />;
```

Comment:
you may have fixed this later in another PR, but does this JSX here actually work? I thought you could use `useNavigation` to get a navigation object and replace the `/login` route imperatively